clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -781,4 +781,25 @@ public static long computeChecksum(ByteBuffer buffer, int start, int size) {
         return Crc32.crc32(buffer.array(), buffer.arrayOffset() + start, size);
     }
 
+    /**
+     * Read data from the channel to the given byte buffer until there are no bytes remaining in the buffer or
+     * the channel has reached end of stream
+     * @param channel File channel containing the data to read from
+     * @param destinationBuffer The buffer into which bytes are to be transferred
+     * @param position The file position at which the transfer is to begin; must be non-negative
+     *
+     * @throws IllegalArgumentException If position is negative
+     * @throws IOException If an I/O error occurs. See {@link FileChannel#read(ByteBuffer, long)}
+     */
+    public static void readFully(FileChannel channel, ByteBuffer destinationBuffer, long position) throws IOException {
+        if (position < 0) {
+            throw new IllegalArgumentException("The file channel position cannot be negative, but we got " + position);
+        }
+        long currentPosition = position;
+        int bytesRead;
+        do {
+            bytesRead = channel.read(destinationBuffer, currentPosition);
+            currentPosition += bytesRead;
+        } while (bytesRead != -1 && destinationBuffer.hasRemaining());
```

Comment:
I think I got what you all mean. Although FileLogInputStream checks whether the buffer has really been filled up after every readFully invoking, it might be possible that other calling code forgets to do this check (i.e. FileRecords.readInto), so we'd better have the readFully return something that can specify if it's successful to read, no matter throwing an exception or returning a boolean flag. Am I correct? 
If yes, I prefer to return a boolean flag to indicating of the state. What do you think?